Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -200,11 +204,12 @@ namespace AzToolsFramework
             }
         }
 
-        void InstanceDomGenerator::UpdateContainerEntityInDomFromRoot(PrefabDom& instanceDom, const Instance& instance) const
+        void InstanceDomGenerator::UpdateContainerEntityInDomFromOwningInstance(PrefabDom& instanceDom, const Instance& instance) const
         {
             // TODO: Modifies the function so it updates the transform only.
 
-            InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(instance, nullptr);
+            const Instance* targetInstance = instance.HasParentInstance() ? &(instance.GetParentInstance()->get()) : nullptr;
+            InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(instance, targetInstance);
```

Comment:
I'm not sure I follow the reasoning behind this change here. It is my understanding that we were previously always fetching the container entity dom 'as seen' by the root dom because we want to account for all the overrides that can possibly be applied on it from any of its ancestors. If it cannot be found in the root prefab, then I'd assume that we would at least try to go 'as close' to the root while climbing up the hierarchy. But the new logic stops at the immediate parent instance.

Reply:
@srikappa-amzn we can change the logic to find the closest container to the root instead of using the immediate parent. Initially I was thinking this is just to get the correct parent so it didn't matter, but if we're copying the whole transform component maybe we need to preserve the other transform values.